internal/indexer/indexer_test.go

```diff
@@ -824,14 +825,55 @@ func TestServiceAccountsByOIDCClaims(t *testing.T) {
 			},
 			expected: []string{"email/fake-email", "email/fake-email-2"},
 		},
+		{
```

Comment:
I would consider renaming the existing tests above here to better reflect what they're now covering. "ServiceAccount has no OIDC email" is probably more properly "has no claim mappings" or something along those lines. "ServiceAccount has OIDC email" could probably stand to be more clear that it's testing the old style of mappings.

On that note, we could probably add a test case below that verifies the index is built correctly for an SA that mixes old and new style mappings.